File: algorithms/blind75/1/binary_search.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        
        lo, hi = 0, len(nums)-1

        while hi >= lo:
            mi = (hi + lo) // 2
            if nums[mi] > target:
                hi = mi - 1
            elif nums[mi] < target:
                lo = mi + 1
            else:
                return mi
        
        return -1

File: algorithms/blind75/1/binary_search.cpp
class Solution {
public:
    int search(vector<int>& nums, int target) {
        int hi = size(nums)-1;
        int lo = 0;
        int mi = 0;
        while (hi >= lo) {
            mi = (hi + lo) / 2;
            if (nums[mi] == target) {
                return mi;
            } else if (nums[mi] > target) {
                hi = mi - 1;
            } else {
                lo = mi + 1;
            }
        }
        

        return -1;
    }
};
